src/lib.rs

```diff
@@ -346,21 +355,29 @@ fn calculate_waste(
 }
 
 #[inline]
-fn calculate_fee(weight: u32, rate: f32) -> u64 {
-    (weight as f32 * rate).ceil() as u64
+fn calculate_fee(weight: u32, rate: f32) -> Result<u64, FeeError> {
+    if rate <= 0.0 {
+        Err(FeeError::NegativeFeeRate)
+    } else if rate > 1000.0 {
+        Err(FeeError::AbnormallyHighFee)
+    } else {
+        Ok((weight as f32 * rate).ceil() as u64)
+    }
 }
 
 /// Returns the effective value which is the actual value minus the estimated fee of the OutputGroup
 #[inline]
-fn effective_value(output: &OutputGroup, feerate: f32) -> u64 {
-    output
+fn effective_value(output: &OutputGroup, feerate: f32) -> Result<u64, FeeError> {
+    Ok(output
         .value
-        .saturating_sub(calculate_fee(output.weight, feerate))
+        .saturating_sub(calculate_fee(output.weight, feerate)?))
```

Comment:
A better name for this could be `output.spending_weight`.  Weight can imply two things, the weight of the output, or the weight of the spending transaction's input. We need the second weight here.

Reply:
Any input on this?

Reply:
I agree that spending_weight is a better name than weight.  Perhaps input_weight or utxo_weight would be other options to consider.  Anything to differentiate the two would improve the readability of the code.